### ISSUE TITLE
[help] with "Aurora Store"

### ISSUE THREAD

**Issue by throwaway242685** [ImranR98/Obtainium#1659]
it seems like "Aurora Store" has changed the filename date format, and now the day goes first instead of the month.

https://auroraoss.com/AuroraStore/Nightly

Obtainium doesn't seem to detect if there's a new version now.

is there a way to fix this from my side? I mean, when configuring the "Aurora Store" URL settings on Obtainium?

**Comment by DwainZwerg** [comment]
Try it with this RegEx: `AuroraStore-[0-9]+\.[0-9]+\.[0-9]+-nightly-signed-(0?[1-9]|[12][0-9]|3[01])-(0?[1-9]|[1][0-2])-[0-9]\.apk` Does it work then?

**Comment by throwaway26425** [comment]
hey, I think that works, thanks!

**Comment by SanskritFritz** [comment]
That regex only helps with extracting the version from the filename. However I need help with Obtainium not able to find out which apk is the latest version on the [Aurora Store Nightly](https://auroraoss.com/AuroraStore/Nightly) page.
The problem is, that Obtainium sorts the apk links by name and the filenames contain the date in american format: DD-MM-YYYY. I have tried the options "Skip sorting" and "Take first link" and even "Apply version string extraction Regex to entire page" but as promising those options sounded, it does not work. Obtainium finds only AuroraStore-4.5.1-nightly-signed-**01**-07-2024.apk this way, not AuroraStore-4.5.1-nightly-signed-**15**-07-2024.apk (although this should be the first link if the links are not sorted).
@throwaway26425 Can you please reopen this case?

**Comment by throwaway26425** [comment]
yeah, I noticed the other day that I wasn't getting the latest version either.

I just re-opened this again.

**Comment by throwaway26425** [comment]
@ImranR98 sorry for bothering you, but do you have any idea of how we can configure this URL with Obtanium?

https://auroraoss.com/downloads/AuroraStore/Nightly/

**Comment by ImranR98** [comment]
Here's the issue. Even though `20-07-2024` is at the top of the page when it gets rendered in a browser, the actual order of the links in the HTML is different (right click on the page and view source). I guess they get re-ordered by some JS code.

Not sure what the solution is here.

**Comment by SanskritFritz** [comment]
@ImranR98 What about introducing a new Regex/Replace pair that can be used on the link list before they get sorted? I actually hoped the option "_Apply version string extraction Regex to entire page_" does exactly that.

**Comment by SanskritFritz** [comment]
> Here's the issue. Even though `20-07-2024` is at the top of the page when it gets rendered in a browser, the actual order of the links in the HTML is different (right click on the page and view source). I guess they get re-ordered by some JS code.

I'm not sure, I just saved the page to plain html, and searched for apk, the first link was this:
```
        </li><li class="item file">
            <a href="https://auroraoss.com/downloads/AuroraStore/Nightly/AuroraStore-4.5.1-nightly-signed-21-07-2024.apk">
                <span class="icon square"><img src="Nightly%20-%20downloads%20_%20AuroraStore%20_%20Nightly_files/ar-apk.svg" alt="ar-apk"></span>
                <span class="icon landscape"><img src="Nightly%20-%20downloads%20_%20AuroraStore%20_%20Nightly_files/ar-apk.svg" alt="ar-apk"></span>
                <span class="label" title="AuroraStore-4.5.1-nightly-signed-21-07-2024.apk">AuroraStore-4.5.1-nightly-signed-21-07-2024.apk</span>
                <span class="date" data-time="1721522121000">2024-07-21 02:35</span>
                <span class="size" data-bytes="7264213">7.3 MB</span>
            <span class="selector">
            <img src="Nightly%20-%20downloads%20_%20AuroraStore%20_%20Nightly_files/selected.svg" alt="selected">

```

**Comment by DwainZwerg** [comment]
Should fixed by @Hamster45105 https://github.com/ImranR98/apps.obtainium.imranr.dev/issues/179#issuecomment-2299837056

**Comment by SanskritFritz** [comment]
Indeed it works now. Maybe a newer version of Obtainium fixed this? Thanks!

**Comment by DwainZwerg** [comment]
No, the config of Hamster is another than mine with more better Regex.

**Comment by Go-rom** [comment]
Config script by @Hamster45105 has been updated.
It's now based on [a more reliable ressource](https://auroraoss.com/downloads/AuroraStore/Feeds/nightly_feed.json)

See : https://github.com/ImranR98/apps.obtainium.imranr.dev/issues/179#issuecomment-2366964051

I think this issue can be closed now.

**Comment by Flashwalker** [comment]
https://github.com/ImranR98/apps.obtainium.imranr.dev/issues/179#issuecomment-4046777533
